### PR TITLE
refactor(backend): statistics with field resolvers

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest

Use field resolvers to make statistics call performant  in wallet

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: admin/src/graphql/communityStatistics.js [hunk 1]
```diff
@@ -4,12 +4,14 @@ export const communityStatistics = gql`
   query {
     communityStatistics {
       totalUsers
-      activeUsers
       deletedUsers
       totalGradidoCreated
       totalGradidoDecayed
-      totalGradidoAvailable
-      totalGradidoUnbookedDecayed
+      totalAvailable {
```
Comment (ulfgebhardt) [inline admin/src/graphql/communityStatistics.js:10]:
```suggestion
      active {
```
naming is not self explanatory.

**Review by ulfgebhardt — COMMENTED** [review]
Awesome - now we use GraphQL correctly

File: backend/src/graphql/resolver/StatisticsResolver.ts [hunk 2]
```diff
@@ -1,81 +1,113 @@
...
+  }
+
+  @FieldResolver()
+  async totalUsers(): Promise<number> {
+    return await DbUser.count()
+  }
+
+  @FieldResolver()
+  async deletedUsers(): Promise<number> {
+    return (await this.allUsers()) - (await this.totalUsers())
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/StatisticsResolver.ts:34]: Not sure about this:
`````
[2023-01-16T18:12:19.281Z][QUERY]: SELECT COUNT(1) AS `cnt` FROM `users` `User`
[2023-01-16T18:12:19.285Z][QUERY]: SELECT COUNT(1) AS `cnt` FROM `users` `User` WHERE `User`.`deleted_at` IS NULL
[2023-01-16T18:12:19.293Z][QUERY]: SELECT `transaction`.`balance` AS `balance`, `transaction`.`balance_date` AS `balanceDate` FROM `users` `user` INNER JOIN `transactions` `transaction` ON `user`.`id` = `transaction`.`user_id` WHERE ( `transaction`.`balance_date` = (SELECT MAX(t.balance_date) FROM transactions AS t WHERE t.user_id = `user`.`id`) ) AND ( `user`.`deleted_at` IS NULL ) ORDER BY `transaction`.`balance_date` DESC, `transaction`.`id` DESC
[2023-01-16T18:12:22.771Z][QUERY]: SELECT SUM(`transaction`.`amount`) AS totalGradidoCreated FROM `transactions` `transaction` WHERE `transaction`.`type_id` = 1
[2023-01-16T18:12:22.774Z][QUERY]: SELECT SUM(`transaction`.`decay`) AS totalGradidoDecayed FROM `transactions` `transaction` WHERE `transaction`.`decay` IS NOT NULL
```

[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/StatisticsResolver.ts:34]: There is no database call at all, as both values already queried. If I follow your suggestion there will always be a database call

File: admin/src/graphql/communityStatistics.js (hunk 1 above)
Comment (Mogge) [reply admin/src/graphql/communityStatistics.js]: I choose `dynamicStatisticsFields`

File: backend/src/graphql/resolver/StatisticsResolver.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply backend/src/graphql/resolver/StatisticsResolver.ts:34]: ok interesting, that might be valid case which is completely depending on the client query

**Pull request merged**